nova/api/openstack/compute/quota_classes.py

```diff
@@ -57,7 +55,7 @@ def _format_quota_set(self, quota_class, quota_set, filtered_quotas=None,
             result = dict(id=str(quota_class))
         else:
             result = {}
-        original_quotas = copy.deepcopy(self.supported_quotas)
+        original_quotas = list(QUOTAS.resources)
```

Comment:
I don't want the `supported_quotas` holding a reference to an instance of our resources, because we might want to update the resources (and recreate the list during that) dynamically as they depend on things configurable via the API.

Reply:
yeah we discussed this yesterday :)